### DISCUSSION TITLE
Playground page design

### DISCUSSION THREAD

**Discussion by jojojojojoj5564656465465** [open-circle/valibot#893] [Ideas]
Hey there,

Could you enhance the playground usability? Here are a few suggestions:
- Add an event listener to trigger the play button without requiring a mouse click.
- Clear the terminal each time the run button is pressed.
- Implement CSS Grid for a side-by-side layout of the editor and terminal using `grid-template-columns: 2fr 1fr`.

Thanks!”

**Comment by fabian-hiller** [comment]
Hey, thanks for your feedback!

What kind of event listener would you use? At what point should it automatically trigger execution? Maybe a shortcut would be better than an event listener.

Why would you automatically clear the output console? If we did, the user would have no control over it. Maybe a shortcut or an option to configure it would be better then taking control away from the user.

The editor and output console are already displayed side by side. Only on smaller devices like smartphones we use a different layout. Can you provide more details?

**Comment by jojojojojoj5564656465465** [comment]
```js
function handleCtrlEnter(event) {
  if (event.ctrlKey && event.key === 'Enter') {
    document.getElementById('action').click();
  }
}

document.addEventListener('keydown', handleCtrlEnter);
```

CTRL + ENTER => RUn the terminal

**Reply by jojojojojoj5564656465465** [reply]
```js
  useOnWindow(
    'keydown',
    $((event) => {
      if (event.ctrlKey && event.key === 'Enter') {
        runPlayer()
      }
    })
  );```

**Reply by fabian-hiller** [reply]
I think `CTRL + ENTER` or `CMD + ENTER` is a great idea. We could also add `CTRL + BACKSPACE` or `CMD + BACKSPACE` to clear the console. What do you think?

**Reply by fabian-hiller** [reply]
This works now! Thank you for your feedback!

**Comment by jojojojojoj5564656465465** [comment]
![image](https://github.com/user-attachments/assets/1a6401a0-a5ef-497a-87ff-1ff5ca87cb32)

I can't read the error i always need to scroll right

**Reply by fabian-hiller** [reply]
Do you recommend making the output console larger and the code editor smaller?

**Reply by jojojojojoj5564656465465** [reply]
container of both on pc could be larger like width : 90%

**Reply by jojojojojoj5564656465465** [reply]
![image](https://github.com/user-attachments/assets/ee6762b9-9f6a-4b64-a630-497e86293f52)

**Reply by jojojojojoj5564656465465** [reply]
if you can on pc use css grid with 2fr 1fr

**Reply by fabian-hiller** [reply]
We can make it bigger, but we need a limit. Otherwise it becomes unusable for ultra wide screens.

**Reply by jojojojojoj5564656465465** [reply]
max-width:2000px

**Reply by fabian-hiller** [reply]
Thank you very much! As soon as I have more time, I will improve the playground based on your feedback.

**Reply by fabian-hiller** [reply]
I have slightly increased the width of the output terminal on large screens. I will evaluate if we should increase it further in the long run.
